packages/builder/src/helpers/components.ts

```diff
@@ -8,27 +8,32 @@ import {
 } from "@budibase/string-templates"
 import { capitalise } from "@/helpers"
 import { Constants } from "@budibase/frontend-core"
+import { Component, ComponentContext } from "@budibase/types"
 
 const { ContextScopes } = Constants
 
 /**
  * Recursively searches for a specific component ID
  */
-export const findComponent = (rootComponent, id) => {
+export const findComponent = (rootComponent: Component, id: string) => {
   return searchComponentTree(rootComponent, comp => comp._id === id)
 }
 
 /**
  * Recursively searches for a specific component type
  */
-export const findComponentType = (rootComponent, type) => {
+export const findComponentType = (rootComponent: Component, type: string) => {
   return searchComponentTree(rootComponent, comp => comp._component === type)
 }
 
 /**
  * Recursively searches for the parent component of a specific component ID
  */
-export const findComponentParent = (rootComponent, id, parentComponent) => {
+export const findComponentParent = (
+  rootComponent: Component | undefined,
```

Comment:
Could these all be set as optional?

Reply:
rootComponent and id shouldn't be optional (or even allowed to be undefined) but it's just because the code that calls this probably isn't all TS - so we can't guarantee that we're not passing in something undefined by mistake. The 3rd param is optional, since it's intended to be blank when called externally, but it's used when it invokes itself via recursion.

Reply:
As @aptkingston mentioned this is reflecting the current usages... Most of them should not be nullable, but until everything is fully typed will not be safe to refactor